docs/administration-guide/gaffer-deployment/gremlin.md

```diff
@@ -0,0 +1,164 @@
+# Connecting Gremlin to Gaffer
+
+It is possible to use Gremlin as an alternative querying language in Gaffer. To
+make Gremlin available however, there are some additional steps that need to be
+taken to connect to a Gaffer graph via this interface.
+
+## Overview
+
+Gremlin serves as a query layer for a graph that implements the Tinkerpop graph
+structure. As of v2.1.0 Gaffer has made available a library that can be utilised
+to enable Gremlin queries. This library can be included via maven in any project
+using the following dependency definition:
+
+```xml
+<dependency>
+    <groupId>uk.gov.gchq.gaffer</groupId>
+    <artifactId>tinkerpop</artifactId>
+    <version>${gaffer.version}</version>
+</dependency>
+```
+
+The library contains the graph implementation that allows Tinkerpop to talk to a
+Gaffer graph and generally is all that is needed to provide the functionality.
+To actually utilise Gremlin queries however, a connection to whats known as a
+`GraphTraversalSource` is required which is the class from which Gremlin
+traversals are spawned. To do this we recommend utilising the provided
+[Gremlin server framework](https://tinkerpop.apache.org/docs/current/reference/#connecting-gremlin-server)
+which can be configured to use the Gaffer Tinkerpop implementation so that a
+endpoint is available for Gremlin queries.
+
+!!! note
+    For release of Gaffer v2.2.0 a ready made container image will be provided
+    to run a preconfigured Gremlin server for Gaffer.
```

Comment:
Do we have an issue for this on `gaffer-docker` repo?

Reply:
We do [here](https://github.com/gchq/gaffer-docker/issues/334)